plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codemodernizer/CodeTransformUtils.kt

```diff
@@ -214,14 +211,18 @@ suspend fun JobId.pollTransformationStatusAndPlan(
                 }
                 state = newStatus
                 numRefreshes = 0
+                return@waitUntil state
             } catch (e: AccessDeniedException) {
                 if (numRefreshes++ > maxRefreshes) throw e
                 refreshToken(project)
+                return@waitUntil state
             } finally {
                 sleep(sleepDurationMillis)
             }
         }
-    } catch (e: WaiterUnrecoverableException) {
+    } catch (e: Exception) {
```

Comment:
`WaiterUnrecoverableException` means that we ended up in a `failOn` state. 
But other exceptions could happen here like `AccessDeniedException` or other unhandled exceptions. 

I'm not sure we want to hide other possible exceptions here, so how about: 

```
catch (e: Exception) {
  when (ex):
     is WaiterUnrecoverableException, AccessDeniedException -> {what we do now}
     else -> throw
}
```

Reply:
Also could we log the error / stack trace, would ease debugging i recon

Reply:
Changed to throwing exception if it's not WaiterUnrecoverableException or AccessDeniedException.

The logging is already done in `pollUntilJobCompletion`.